ReactWindows/ReactNative.Shared/Bridge/ReactBridge.cs

```diff
@@ -108,11 +110,35 @@ public void Dispose()
 
         private void ProcessResponse(JToken response)
         {
+            ProcessResponse(response, false);
+        }
+
+        private void ProcessResponse(JToken response, bool isEndOfBatch)
+        {
+            bool localBatchHadNativeModuleCalls;
+
             if (response == null || response.Type == JTokenType.Null || response.Type == JTokenType.Undefined)
             {
+                if (isEndOfBatch)
+                {
+                    localBatchHadNativeModuleCalls = _batchHadNativeModuleCalls;
+
+                    _nativeModulesQueueThread.Dispatch(() =>
+                    {
+                        // the bridge could still be processing native calls when the idle signaler fires
+                        if (localBatchHadNativeModuleCalls)
+                        {
+                            _reactCallback.OnBatchComplete();
+                        }
+                    });
+
+                    _batchHadNativeModuleCalls = false;
+                }
+
```

Comment:
This is overly complicated. Are we overthinking what the `response` value can be? Is it ever going to be `null` or `undefined`?

Reply:
I imagine it's always an array.

---
In reply to: [181256506](https://github.com/Microsoft/react-native-windows/pull/1769#discussion_r181256506) [](ancestors = 181256506)